test/http.cpp

```diff
@@ -415,6 +418,30 @@ void test_large_request() {
   client.batch();
 }
 
+void test_health_check() {
```

Comment:
the changes in the tests are straight forward. use the new options and test that they work